src/index.js

```diff
@@ -25,7 +26,11 @@ Walker.prototype._read = function () {
   self.fs.lstat(pathItem, function (err, stats) {
     var item = { path: pathItem, stats: stats }
     if (err) return self.emit('error', err, item)
-    if (!stats.isDirectory()) return self.push(item)
+
+    if (!stats.isDirectory() || (self.rootDepth &&
+      path.normalize(pathItem).split(path.sep).length - self.rootDepth >= self.options.depthLimit)) {
```

Comment:
@gsf-sellis same here. No need to use `path.normalize(pathItem)` since `pathItem` already normalized either by `path.resolve()` or `path.join()`. Thanks!